File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpPlaceBoxRequest.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.Map;

public record RMcpPlaceBoxRequest(
        String blockId,
        RMcpBlockPosData startPos,
        RMcpBlockPosData endOffset,
        Map<String, String> state,
        boolean dryRun
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpPlaceBoxRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpPlaceBoxRequest(
    val blockId: String,
    val startPos: RBlockPos,
    val endOffset: RBlockPos,
    val state: Map<String, String>,
    val dryRun: Boolean
)
